core/src/main/java/org/hisp/dhis/android/core/trackedentity/search/TrackedEntityInstanceQueryOnlineHelper.kt

```diff
@@ -172,11 +175,13 @@ internal class TrackedEntityInstanceQueryOnlineHelper @Inject constructor(
         }
 
         private fun getAPIValue(item: RepositoryScopeFilterItem): String {
+            val charsToEscape = "[;,:]".toRegex()
+
             return if (item.operator() == FilterItemOperator.IN) {
-                val list = FilterOperatorsHelper.strToList(item.value())
+                val list = FilterOperatorsHelper.strToList(item.value()).map { it.replace(charsToEscape, "/\$0") }
                 list.joinToString(";")
             } else {
-                item.value()
+                item.value().replace(charsToEscape, "/\$0")
```

Comment:
The string `/\$0` is duplicated. It can be moved to a function such as `escapeAPIValue(value: String): String`.